### ISSUE TITLE
PyTorch without distributed support not supported

### ISSUE THREAD

**Issue by adamjstewart** [lightly-ai/lightly#1153]
If PyTorch isn't built with distributed support (commonly the case on macOS), many features of lightly crash with an error. For example:
```
lib/python3.10/site-packages/lightly/loss/ntx_ent_loss.py:164: in forward
    labels = labels + dist.rank() * batch_size
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def rank() -> int:
        """Returns the rank of the current process."""
>       return dist.get_rank() if dist.is_initialized() else 0
E       AttributeError: module 'torch.distributed' has no attribute 'is_initialized'
```
Any time `torch.distributed` is used, it should be wrapped by:
```python
if torch.distributed.is_available():
```
Lightning themselves have introduced and fixed this same bug several times:

* https://github.com/Lightning-AI/lightning/issues/16637
* https://github.com/Lightning-AI/lightning/issues/15494
* https://github.com/Lightning-AI/lightning/issues/10348

**Comment by guarin** [comment]
Ah, thanks for bringing this up! We use dist in multiple places and should definitely check for it.

**Comment by guarin** [comment]
This should be fixed with #1180.
